pydy/bodies.py

```diff
@@ -0,0 +1,158 @@
+    >>> from sympy import Point
+    >>> from pydy.bodies import Body
+    >>> mass = Symbol('mass')
+    >>> masscenter = Symbol('masscenter')
+    >>> body = Body('name_of_body', masscenter, mass)
+
+    Similarly, A frame can also be passed while creating a Particle.
+
+    """
+    def __init__(self, name, masscenter=None, mass=None, frame=None,
+                 inertia=None):
+
+        self.name = name
+        self.parent = None
+        self.child = None
+        self.force_list = []
+        self.body = None  # TODO assign a better name.
+        self._coordinates = []
+        self._speeds = []
+        self._counter = 0
+
+        if masscenter is None:
+            self._masscenter = Point(self.name + '_masscenter')
+        else:
+            self._masscenter = masscenter
+
+        if mass is None:
+            self._mass = Symbol(self.name + '_mass')
+        else:
+            self._mass = mass
+
+        if frame is None:
+            self._frame = ReferenceFrame(self.name + '_frame')
+        else:
+            self._frame = frame
+
+        if inertia is None and mass is None:
+            inertia = outer(self._frame.x, self._frame.x)  # tensor product
+            self._inertia = (inertia, self._masscenter)
+        else:
+            self._inertia = inertia
+
+        # If user passes masscenter and mass then a particle is created
+        # otherwise a rigidbody. As a result a body may or may not have inertia.
+        if inertia is None and mass is not None:
+            self.body = Particle(self.name, self._masscenter, self._mass)
+        else:
+            self.body = RigidBody(self.name, self._masscenter, self._frame,
+                                  self._mass, self._inertia)
+
+    def add_coordinate(self, coordinate):
+        self._coordinates.append(coordinate)
+
+    def add_speed(self, speed):
+        self._speeds.append(speed)
+
+    def add_force(self, point_vector, force_vector):
+        """
+        Adds force to the body by adding Force's instance to the force_list.
+        force_list is used by system to get the tuples from the instance and
+        create force_list for Kane's method.
+
+        Parameters
+        ---------
+        point_vector: A 3 Tuple
+            Defines an arbitary point on a body on which the force will be
+            applied. The tuple defines the x, y and z values of the point
+            w.r.t body's masscenter.
+        force_vector: A 3 Tuple
+            Defines the force vector w.r.t frame of the body. The tuple defines
+            the values of vector in x, y and z directions of body's frame.
+
+        Example
+        --------
+        To add a force of magnitude 1 in y direction on a point at distance of
+        1 in x direction. All the directions are w.r.t body's frame.
+
+        >>> body = Body('body')
+        >>> body.add_force((1,0,0), (0,1,0))
+
+        """
+        point_vector = self._convert_tuple_to_vector(point_vector)
```

Comment:
You should put in some checks here, to see if users are supplying the tuple. 
The purpose of this check is to provide a more informative error/tracebacks. 
something like 'This argument should be a tuple..'